hexl/fft/fft-native.cpp

```diff
@@ -0,0 +1,410 @@
+// Copyright (C) 2020-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "hexl/fft/fft-native.hpp"
+
+#include <cstring>
+
+#include "util/util-internal.hpp"
+
+namespace intel {
+namespace hexl {
+
+inline void ComplexFwdButterflyRadix2(std::complex<double>* X_r,
+                                      std::complex<double>* Y_r,
+                                      const std::complex<double>* X_op,
+                                      const std::complex<double>* Y_op,
+                                      const std::complex<double> W) {
+  std::complex<double> U = *X_op;
+  std::complex<double> V = *Y_op * W;
+  *X_r = U + V;
+  *Y_r = U - V;
+}
+
+inline void ComplexInvButterflyRadix2(std::complex<double>* X_r,
+                                      std::complex<double>* Y_r,
+                                      const std::complex<double>* X_op,
+                                      const std::complex<double>* Y_op,
+                                      const std::complex<double> W) {
+  std::complex<double> U = *X_op;
+  *X_r = U + *Y_op;
+  *Y_r = (U - *Y_op) * W;
+}
+
+inline void ScaledComplexInvButterflyRadix2(std::complex<double>* X_r,
+                                            std::complex<double>* Y_r,
+                                            const std::complex<double>* X_op,
+                                            const std::complex<double>* Y_op,
+                                            const std::complex<double> W,
+                                            const double* scalar) {
+  std::complex<double> U = *X_op;
+  *X_r = (U + *Y_op) * (*scalar);
+  *Y_r = (U - *Y_op) * W;
+}
```

Comment:
These functions in the equivalent NTT functions have HEXL macros performing checks

Reply:
Do you mean HEXL_VLOG?